Code/Framework/AzFramework/AzFramework/Windowing/NativeWindow.cpp

```diff
@@ -90,82 +97,170 @@ namespace AzFramework
     }
 
     bool NativeWindow::SupportsClientAreaResize() const
     {
-        return m_pimpl->SupportsClientAreaResize();
+        if (m_pimpl != nullptr)
+        {
+            return m_pimpl->SupportsClientAreaResize();
+        }
+        else
+        {
+            return false;
+        }
     }
 
     void NativeWindow::SetEnableCustomizedResolution(bool enable)
     {
-        m_pimpl->SetEnableCustomizedResolution(enable);
+        if (m_pimpl != nullptr)
+        {
+            m_pimpl->SetEnableCustomizedResolution(enable);
+        }
     }
 
     bool NativeWindow::IsCustomizedResolutionEnabled() const
     {
-        return m_pimpl->IsCustomizedResolutionEnabled();
+        if (m_pimpl != nullptr)
+        {
+            return m_pimpl->IsCustomizedResolutionEnabled();
+        }
+        else
+        {
+            return false;
+        }
     }
 
     WindowSize NativeWindow::GetRenderResolution() const
     {
-        return m_pimpl->GetRenderResolution();
+        if (m_pimpl != nullptr)
+        {
+            return m_pimpl->GetRenderResolution();
+        }
+        else
+        {
+            return WindowSize();
+        }
     }
 
     void NativeWindow::SetRenderResolution(WindowSize resolution)
     {
-        m_pimpl->SetRenderResolution(resolution);
+        if (m_pimpl != nullptr)
+        {
+            m_pimpl->SetRenderResolution(resolution);
+        }
     }
 
     bool NativeWindow::GetFullScreenState() const
     {
-        return m_pimpl->GetFullScreenState();
+        if (m_pimpl)
+        {
+            return m_pimpl->GetFullScreenState();
+        }
+        else
+        {
+            return false;
+        }
     }
 
     void NativeWindow::SetFullScreenState(bool fullScreenState)
     {
-        m_pimpl->SetFullScreenState(fullScreenState);
+        if (m_pimpl)
+        {
+            m_pimpl->SetFullScreenState(fullScreenState);
+        }        
     }
 
     bool NativeWindow::CanToggleFullScreenState() const
     {
-        return m_pimpl->CanToggleFullScreenState();
+        if (m_pimpl)
+        {
+            return m_pimpl->CanToggleFullScreenState();
+        }
+        else
+        {
+            return false;
+        }
     }
 
     void NativeWindow::ToggleFullScreenState()
     {
-        SetFullScreenState(!GetFullScreenState());
+        if (m_pimpl)
+        {
+            SetFullScreenState(!GetFullScreenState());
+        }
     }
 
     float NativeWindow::GetDpiScaleFactor() const
     {
-        return m_pimpl->GetDpiScaleFactor();
+        if (m_pimpl)
+        {
+            return m_pimpl->GetDpiScaleFactor();
+        }
+        else
+        {
+            return 1.0f;
+        }
     }
 
     uint32_t NativeWindow::GetDisplayRefreshRate() const
     {
-        return m_pimpl->GetDisplayRefreshRate();
+        if (m_pimpl)
+        {
+            return m_pimpl->GetDisplayRefreshRate();
+        }
+        else    
+        {
+            return 0;
+        }
```

Comment:
repeat: This logic can be coalesced into a ternary statement
```suggestion
        return m_pimpl != nullptr ? m_pimpl->GetDisplayRefreshRate() : 0;
```